src/clients/victoriaMetrics/promAgent.ts

```diff
@@ -0,0 +1,102 @@
+import promClient from 'prom-client';
+import { sendToDiscordAdmins } from '../discord';
+
+const MAX_PUSH_METRICS_ERRORS = 5;
+const PUSH_METRICS_ERRORS_RATE_LIMIT = 60 * 60 * 1000; // 1 hour
+
+export class PromAgent {
+  private metricNames: Set<string>;
+  private pushMetricsErrorsTimestamps: number[] = [];
+
+  constructor() {
```

Comment:
Let's add a function handler here in the constructor for this class to call when a duplicate metric is registered. 

Makes it easier for testing and makes the code cleaner.